src/lib/y2storage/proposal/autoinst_partition_size.rb

```diff
@@ -0,0 +1,74 @@
+# encoding: utf-8
+
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Y2Storage
+  module Proposal
+    # Mixin with methods used to calculate the min size, max size and weight of
+    # a set of partitions which sizes are initially specified as AutoYaST sizes.
+    module AutoinstPartitionSize
+      # Returns a list of planned partitions adjusting the size
+      #
+      # All partitions which sizes are specified as percentage will get their minimal and maximal
+      # sizes adjusted.
+      #
+      # If a device is provided as argument, it is considered that the
+      # partitions will be all created in such device. If not, a devicegraph
+      # must be provided and the device for each partition will be searched in
+      # the devicegraph using the corresponding #{Planned::Partition#disk}
+      # attribute.
```

Comment:
np: From my POV, having two named arguments which are exclusive is a little bit confusing.

Reply:
I agree. But being just a mixin for kind of internal usage, I considered it acceptable.